### PR TITLE
Add bulk transformation operations support

### PR DESCRIPTION
**Pull request by frankbria**
Implement bulk transformation operations that allow users to apply the same transformation to multiple columns simultaneously with progress tracking.

Backend changes:
- Add BulkTransformationJob model with progress tracking and job management
- Add ColumnSelectionPattern model for pattern-based column selection
- Create BulkTransformationService with parallel column processing
- Add API endpoints for column selection, bulk preview, and bulk apply
- Add 28 unit tests for the bulk transformation service

Frontend changes:
- Extend TransformationService with bulk operation API methods
- Add BulkColumnSelector component with Shift+click range selection
- Add SelectionPatternBuilder for pattern-based column selection
- Add BulkTransformationPanel for operation configuration
- Add BulkPreview component for multi-column preview display
- Add BulkTransformationProgress component with real-time updates
- Add frontend unit tests for BulkColumnSelector

Features:
- Select columns by data type, name pattern, or quality metrics
- Preview transformations on multiple columns before applying
- Parallel processing with configurable concurrency limits
- Real-time progress tracking with estimated time remaining
- Stop-on-error option for error handling control
- Job cancellation and retry support

Closes Issue #70 

## Summary by CodeRabbit

* **New Features**
  * Bulk transformation workflow: pattern-based column selection, per-column preview, apply-as-background-job with job listing, status polling, cancel/retry, and real-time progress UI.
  * New frontend components and client APIs to configure, preview and run bulk operations.
  * Backend job model and service enabling asynchronous execution, progress tracking, results persistence and retry behavior.

* **Bug Fixes**
  * Standardized error handling and status mapping for bulk endpoints (validation, not-found, rate limits).

* **Tests**
  * Extensive unit and UI tests covering selection, preview, job lifecycle, validation and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by frankbria** [comment]
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scanner |  Critical |  High |  Medium |  Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/frankbria/pr-checks/f4a8946b-c4ec-4c17-9fec-996265d683c8?refs=b4b3c20e5e8136624c1c57fb0ace741c41999833&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

File: apps/frontend/__tests__/components/BulkColumnSelector.test.tsx [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+      <BulkColumnSelector
+        columns={[]}
+        selectedColumns={new Set()}
+        onSelectionChange={mockOnSelectionChange}
+        error="Failed to load columns"
+      />
+    );
+
+    expect(screen.getByText('Failed to load columns')).toBeInTheDocument();
+  });
```
[the remark this answers is not in the document]
Comment (frankbria) [reply apps/frontend/__tests__/components/BulkColumnSelector.test.tsx]: Fixed

File: apps/backend/app/services/bulk_transformation_service.py [hunk 2]
```diff
@@ -0,0 +1,713 @@
...
+                """Process a single column with semaphore."""
+                async with semaphore:
+                    return await self._transform_column(
+                        df=df,
+                        column=column,
+                        transformation_type=job.transformation_type,
+                        global_parameters=job.global_parameters,
+                        per_column_params=job.per_column_params,
+                    )
+
```
[the remark this answers is not in the document]
Comment (frankbria) [reply apps/backend/app/services/bulk_transformation_service.py]: Fixed
